integration/gateway/gateway_test.go

```diff
@@ -201,17 +209,77 @@ var _ = Describe("GatewayService", func() {
 			_, err = gatewayClient.Submit(ctx, submitRequest)
 			Expect(err).NotTo(HaveOccurred())
 
+			identity, err = signingIdentity.Serialize()
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			conn.Close()
+			cancel()
+		})
+
+		It("should respond with status of submitted transaction", func() {
 			statusRequest := &gateway.CommitStatusRequest{
 				ChannelId:     "testchannel",
+				Identity:      identity,
 				TransactionId: transactionID,
 			}
-			actualStatus, err := gatewayClient.CommitStatus(ctx, statusRequest)
+
+			statusRequestBytes, err := proto.Marshal(statusRequest)
+			Expect(err).NotTo(HaveOccurred())
+
+			signedStatusRequest := &gateway.SignedCommitStatusRequest{
+				Request: statusRequestBytes,
+			}
+			signedStatusRequest.Signature, err = signingIdentity.Sign(statusRequestBytes)
+			Expect(err).NotTo(HaveOccurred())
+
+			actualStatus, err := gatewayClient.CommitStatus(ctx, signedStatusRequest)
 			Expect(err).NotTo(HaveOccurred())
 
 			expectedStatus := &gateway.CommitStatusResponse{
 				Result: peer.TxValidationCode_VALID,
 			}
 			Expect(proto.Equal(actualStatus, expectedStatus)).To(BeTrue(), "Expected\n\t%#v\nto proto.Equal\n\t%#v", actualStatus, expectedStatus)
 		})
+
+		It("should fail on invalid identity", func() {
+			statusRequest := &gateway.CommitStatusRequest{
+				ChannelId:     "testchannel",
+				Identity:      []byte("BAD_IDENTITY"),
+				TransactionId: transactionID,
+			}
+
+			statusRequestBytes, err := proto.Marshal(statusRequest)
+			Expect(err).NotTo(HaveOccurred())
+
+			signedStatusRequest := &gateway.SignedCommitStatusRequest{
+				Request: statusRequestBytes,
+			}
+			signedStatusRequest.Signature, err = signingIdentity.Sign(statusRequestBytes)
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = gatewayClient.CommitStatus(ctx, signedStatusRequest)
+			Expect(err).To(HaveOccurred())
```

Comment:
In error checks, its desired to check for specific error (response code or error message - whatever makes it unique) so as to make sure that the test does not pass for a wrong reason.